diff --git a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/android/cpp-adapter.cpp b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/android/cpp-adapter.cpp
index d8d1c1a..41172a0 100644
--- a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/android/cpp-adapter.cpp
+++ b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/android/cpp-adapter.cpp
@@ -29,34 +29,14 @@ JNIEXPORT jboolean JNICALL
     jlong rtPtr,
     jobject callInvokerHolderJavaObj
 ) {
-    // https://github.com/realm/realm-js/blob/main/packages/realm/binding/android/src/main/cpp/io_realm_react_RealmReactModule.cpp#L122-L145
-    // React Native uses the fbjni library for handling JNI, which has the concept of "hybrid objects",
-    // which are Java objects containing a pointer to a C++ object. The CallInvokerHolder, which has the
-    // invokeAsync method we want access to, is one such hybrid object.
-    // Rather than reworking our code to use fbjni throughout, this code unpacks the C++ object from the Java
-    // object `callInvokerHolderJavaObj` manually, based on reverse engineering the fbjni code.
-
-    // 1. Get the Java object referred to by the mHybridData field of the Java holder object
-    auto callInvokerHolderClass = env->GetObjectClass(callInvokerHolderJavaObj);
-    auto hybridDataField = env->GetFieldID(callInvokerHolderClass, "mHybridData", "Lcom/facebook/jni/HybridData;");
-    auto hybridDataObj = env->GetObjectField(callInvokerHolderJavaObj, hybridDataField);
-
-    // 2. Get the destructor Java object referred to by the mDestructor field from the myHybridData Java object
-    auto hybridDataClass = env->FindClass("com/facebook/jni/HybridData");
-    auto destructorField =
-        env->GetFieldID(hybridDataClass, "mDestructor", "Lcom/facebook/jni/HybridData$Destructor;");
-    auto destructorObj = env->GetObjectField(hybridDataObj, destructorField);
-
-    // 3. Get the mNativePointer field from the mDestructor Java object
-    auto destructorClass = env->FindClass("com/facebook/jni/HybridData$Destructor");
-    auto nativePointerField = env->GetFieldID(destructorClass, "mNativePointer", "J");
-    auto nativePointerValue = env->GetLongField(destructorObj, nativePointerField);
-
-    // 4. Cast the mNativePointer back to its C++ type
-    auto nativePointer = reinterpret_cast<facebook::react::CallInvokerHolder*>(nativePointerValue);
-    auto jsCallInvoker = nativePointer->getCallInvoker();
+    using JCallInvokerHolder = facebook::react::CallInvokerHolder;
 
+    auto holderLocal = facebook::jni::make_local(callInvokerHolderJavaObj);
+    auto holderRef = facebook::jni::static_ref_cast<JCallInvokerHolder::javaobject>(holderLocal);
+    auto* holderCxx = holderRef->cthis();
+    auto jsCallInvoker = holderCxx->getCallInvoker();
     auto runtime = reinterpret_cast<jsi::Runtime *>(rtPtr);
+
     return {{ ns }}::installRustCrate(*runtime, jsCallInvoker);
 }
 
diff --git a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/ios/module-template.podspec b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/ios/module-template.podspec
index 3de6fbc..1797e41 100644
--- a/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/ios/module-template.podspec
+++ b/node_modules/uniffi-bindgen-react-native/crates/ubrn_cli/templates/jsi/ios/module-template.podspec
@@ -30,7 +30,7 @@ Pod::Spec.new do |s|
   {%- let tm = self.relative_to(root, dir) %}
   {%- let dir = self.config.project.bindings.cpp_path(root) %}
   {%- let bindings = self.relative_to(root, dir) -%}
-  s.source_files = "{{ ios }}/**/*.{h,m,mm,swift}", "{{ codegen }}/**/*.{h,m,mm}", "{{ tm }}/**/*.{hpp,cpp,c,h}", "{{ bindings }}/**/*.{hpp,cpp,c,h}"
+  s.source_files = "{{ ios }}/*.{h,m,mm,swift}", "{{ codegen }}/**/*.{h}", "{{ tm }}/**/*.{hpp,cpp,c,h}", "{{ bindings }}/**/*.{hpp,cpp,c,h}"
   s.vendored_frameworks = "{{ framework }}"
   s.dependency    "uniffi-bindgen-react-native", "{{ self.config.project.ubrn_version() }}"
 
